kafka-client/src/main/java/dev/responsive/kafka/store/ResponsiveMaterialized.java

```diff
@@ -17,14 +17,27 @@ public ResponsiveMaterialized(
   ) {
     super(materialized);
     this.truncateChangelog = truncateChangelog;
+
+    // Truncation uses the admin's delete request, which seems to require the cleanup.policy
+    // to contain 'delete'
     if (truncateChangelog) {
-      topicConfig.put(TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_DELETE);
+      topicConfig.put(
+          TopicConfig.CLEANUP_POLICY_CONFIG,
+          TopicConfig.CLEANUP_POLICY_COMPACT + "," + TopicConfig.CLEANUP_POLICY_DELETE
```

Comment:
Small change here -- when truncation _is_ enabled, I think it makes sense to configure it with `compact, delete` rather than just `delete`, since the default is `compact`. And at this point, only the default configs are enabled -- to actually change any configs you have to go through the `withLoggingEnabled` API, where we have a separate validation but otherwise let users override it as they please

Note that these configs are only applied if the changelog is created anew. For existing changelogs, if they are not `delete` enabled, we will just fail the `deleteRecords` request in the CommitBuffer and then turn off further truncation attempts.

Reply:
if there is not existing topic why not just do the most sensible thing for our store? with just `delete` there's zero risk of losing tombstones right? What's the benefit of staying consistent with how vanilla streams would configure the changelog?

Reply:
I actually do think `delete, compact` _is_ what we want for our stores, with changelog truncation enabled. Here's how I was reasoning about it, but definitely think about it and whether you agree:

1. I think the benefit is undeniable, though perhaps negligible outside of certain edge cases. But imagine an app with mostly all overwrites of the same key. If we end up crashing/failing the flush/getting fenced/etc and there's a large buildup of single-key overwrites, with just `delete` we'll have to consume and write all of these as the restoration path doesn't deduplicate records in any way (like our CommitBuffer does). Whereas with `delete, compact` we only have to restore the latest value (per key), which is the same as what our CommitBuffer would be trying to flush had it gone through normal processing instead of restoration. I'm worried we could get cascading failures (maybe a bit extreme but you get the idea) if the data to be written via restoration is potentially orders of magnitude larger than the write from a CommitBuffer flush
Similarly, in the case of repetitive commit failures under ALOS, we'll rewind to the point of last flush/commit and then rewrite everything in the changelog each time before attempting the (doomed) commit flush. In that case, without compaction we'd be digging a deeper and deeper grave as each failure meant another full set of data that need to be restored and written to Cassandra (even though it's actually just repeated sequences)
2. I believe the risk is negligible, because we aggressively truncate after each flush which means the changelog only ever contains the current "working set" of data since the last successful flush/commit. It's hard to imagine an application that could be successful with a flush interval on the order of 24 hours, which is how long tombstones are kept around for by default (and we could override/validate this config to be sure, or even set it to 7 days to feel extra safe).
3. Somewhat orthogonal, I'm a little worried that users might enable changelog truncation without realizing that it means turning on time-based topic retention, and could lose data if their application has long downtimes beyond the default of 7 days. I'm wondering if we should actually set an infinite retention time, and simply rely on the truncation and compaction semantics to keep the changelog topic growth in check